components/tidb_query/src/batch/executors/stream_aggr_executor.rs

```diff
@@ -247,30 +254,47 @@ impl<Src: BatchExecutor> AggregationExecutorImpl<Src> for BatchStreamAggregation
             for group_by_result in &self.group_by_results_unsafe {
                 group_key_ref.push(group_by_result.get_logical_scalar_ref(logical_row_idx));
             }
-            match self.keys.rchunks_exact(group_by_len).next() {
-                Some(current_key) if &group_key_ref[..] == current_key => {
-                    group_key_ref.clear();
-                }
-                _ => {
-                    // Update the complete group
-                    if logical_row_idx > 0 {
-                        update_current_states(
-                            context,
-                            &mut self.states,
-                            aggr_fn_len,
-                            &self.aggr_expr_results_unsafe,
-                            group_start_logical_row,
-                            logical_row_idx,
-                        )?;
+            let group_match = 'group_match: {
```

Comment:
I think a closure + return can avoid using this unstable feature.

Reply:
done